pkg/types/labels.go

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package types
+
+import "time"
+
+type Label struct {
```

Comment:
We should consider if we want to do some restrictions regarding what symbols can be contained in label keys and values and whatever we decide should be introduced with this change so that we can avoid having label migrations afterwards

Reply:
If an entity doesn't support labels, the storage won't implement the additional logic regarding labels.

Reply:
I was thinking for some general validation that would apply for all label types later on - if we can think of any right now, we should probably implement it now before we have stored labels

Reply:
at best it can be extracted in this struct, but every other label that embeds it, should call validate